### PR TITLE
Hasp & Cryptic

### PR DESCRIPTION
**Pull request by davidcp90**
Included 2 additional utilities
Cryptic: Encrypt and decrypt objects

```
import { encrypt, decrypt } from '@mercadoni/elementals/cryptic'
const encryptionKey = 'choosenEncryptionKey'
const objToEncrypt = { id: 1, name: 'hi }
const encrypted = encrypt(encriptionKey,objToEncrypt) // encrypt the original object
decrypt(encryptionKey, encrypted) // returns the original object

```
Hasp: Get the client ID given a configuration, or deny access
```
import { getClient } from 'hasp';

const configObject = {
 beautifulKey: 'CLIENTID'
}
const clientMiddleware = getClient(configObject, true) //set second param as false if you dont want logs from this middleware
const yourAppRouter = Router();

yourAppRouter.use(clientMiddleware);
```

### CODE REVIEW

File: src/cryptic/index.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const crypto = require('crypto')
+const algorithm = 'aes-256-cbc'
```
Comment (cerodriguezl) [inline src/cryptic/index.ts:2]: This could be a parameter passed to the function

File: src/cryptic/index.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const iv = crypto.randomBytes(16)
```
Comment (cerodriguezl) [inline src/cryptic/index.ts:3]: I'm not sure if this is instantiated just once, when the package is imported. If this is some kind of seed, it is safer to get a new one every time the encrypt method is called

File: src/cryptic/index.ts (hunk 1 above)
Comment (davidcp90) [reply src/cryptic/index.ts:2]: Good idea

File: src/cryptic/index.ts (hunk 2 above)
Comment (davidcp90) [reply src/cryptic/index.ts:3]: Yes, it's  a seed , I'd do the change

**Pull request merged**